File: Frontend/Song/src/app/song.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';

@Injectable({
  providedIn: 'root'
})
export class SongService {

  baseUrl = "http://localhost:8085/api/v1";
  constructor(private http:HttpClient) { }

  getAllSong()
  {
    return this.http.get(this.baseUrl+"/get-all-song");
  }

  addSong(songobj:any)
  {
    return this.http.post(this.baseUrl+"/add-song", songobj);
  }

  deleteSong(songId:any)
  {
    return this.http.delete(this.baseUrl+"/song/" + songId);
  }

  updateSong(songId:any, songobj:any)
  {
    return this.http.put(this.baseUrl+"/song/" + songId, songobj);
  }
}

File: Backend/song/src/main/java/com/project/song/service/SongService.java
package com.project.song.service;

import com.project.song.model.Songs;

import java.util.List;

public interface SongService
{
    public abstract Songs saveSong(Songs songs);
    public abstract List<Songs> getAllSong();
    public abstract Songs updateSong(Songs songs, String songId);
    public abstract boolean deleteSong(String songId);
}
